Dockerfile

```diff
@@ -0,0 +1,51 @@
+FROM ubuntu:trusty
+MAINTAINER Di Wu <diwu@yelp.com>
+
+COPY . /opt/redash/current/
+
+WORKDIR /opt/redash/current
+
+ENV FILES_BASE_URL /opt/redash/current/setup/files/
+
+# Base packages
+RUN apt-get update && \
+  apt-get install -y python-pip python-dev curl build-essential pwgen libffi-dev sudo git-core
+RUN pip install -U setuptools
+
+# redash user
+RUN useradd --system --comment " " --create-home redash
+RUN useradd --system --comment " " --create-home postgres
+
+# Install dependencies via apt-get
+RUN apt-get update && \
+  apt-get -y install libpq-dev postgresql-client
+
+# Make logs folder
+RUN mkdir /opt/redash/logs
+
+# Default config file
+RUN cp $FILES_BASE_URL"env" /opt/redash/.env
+
+# Install dependencies
+RUN cd /opt/redash/current
+RUN pip install -r requirements.txt
+
+RUN apt-get install -y libssl-dev && \
+  apt-get install -y libmysqlclient-dev
+
+# Pip requirements for all data source types
+RUN pip install -r requirements_all_ds.txt
+
+# Setup supervisord + sysv init startup script
+RUN mkdir -p /opt/redash/supervisord
+RUN pip install supervisor==3.1.2
+
+# Get supervisord startup script
+RUN cp $FILES_BASE_URL"supervisord_docker.conf" /opt/redash/supervisord/supervisord.conf
+
+RUN cp $FILES_BASE_URL"redash_supervisord_init" /etc/init.d/redash_supervisord
+
+# Fix permissions
+RUN chown -R redash /opt/redash
+
+CMD bash /etc/init.d/redash_supervisord start
```

Comment:
Ok, instead of commenting on separate lines, I reviewed the whole file and here's what I suggest. Note that this is only a suggestion.

```Dockerfile
FROM ubuntu:trusty
MAINTAINER Di Wu <diwu@yelp.com>

# Ubuntu packages
RUN apt-get update && \
    apt-get install -y python-pip python-dev curl build-essential pwgen libffi-dev sudo git-core && \
    # Postgres client
    apt-get -y install libpq-dev postgresql-client && \
    # Additional packages required for data sources:
    apt-get install -y libssl-dev libmysqlclient-dev

ENV REDASH_STATIC_ASSETS_PATH="../rd_ui/dist/" \
    REDASH_LOG_LEVEL="INFO" \
    REDASH_REDIS_URL=redis://redis:6379/0 \
    REDASH_DATABASE_URL="postgresql://redash@postgres/redash"

# Get latest version tarball:
# We install it directly into /opt/redash/current rather than symlink, because in 
# a Docker environment we will replace the re:dash image, rather than upgrade it.
RUN wget $(python latest_release_url.py) -O redash.tar.gz && \
    mkdir -p /opt/redash/current && \
    tar -C /opt/redash/current -xvf redash.tar.gz && \
    rm redash.tar.gz

# Install Python dependencies
WORKDIR /opt/redash/current
RUN pip install -U setuptools && \
    pip install -r requirements.txt && \
    # pip requirements for all data source types
    pip install -r requirements_all_ds.txt

# Setup supervisord
RUN mkdir -p /opt/redash/supervisord && \
    mkdir -p /opt/redash/logs && \
    pip install supervisor==3.1.2 && \
    cp /opt/redash/current/setup/files/supervisord_docker.conf /opt/redash/supervisord/supervisord.conf

# Get latest version tarball
CMD ["supervisord -c /opt/redash/supervisord/supervisord.conf"]
```

Notes:
1. I grouped all apt-get commands from across the file.
2. I removed user creation: we don't need the postgres user, it exists in the postgres image. Also we don't really need the redash user, as it less needed in Docker environment and will only complicate things (we will run things as root; we can revisit this later on).
3. Instead of `.env` file, let's use ENV directive and `env_file`/`env` option in docker-compose. We need to remember to pass REDASH_COOKIE_SECRET in docker-compose. Also we need to update the supervisord config to not use bin/run.
4. latest_release_url.py is a script that will get the url of the latest tarball file (see below).
5. I'm not sure how the Docker postgres image handles authentication, so the database URL might need to be updated. Also we need to update the `docker_init_postgres.sh` script to run from the context of the docker host, and run some commands with the redash image and some with the postgres image.
6. I suggest to move the Dockerfile and its related scripts/files into `setup/docker`.

```python
# latest_release_url.py
import urllib2
import json

latest = json.load(urllib2.urlopen("https://api.github.com/repos/EverythingMe/redash/releases/latest"))

print latest['assets'][0]['browser_download_url']
```